File: frontend/lib/models/debt.dart
class Debt {
  final String id;
  final String consumerName;
  final List<DebtItem> items;
  final double totalAmount;
  final String status;
  final DateTime createdAt;
  final String? branchId;
  final String? businessId;

  Debt({
    required this.id,
    required this.consumerName,
    required this.items,
    required this.totalAmount,
    required this.status,
    required this.createdAt,
    this.branchId,
    this.businessId,
  });

  factory Debt.fromJson(Map<String, dynamic> json) {
    return Debt(
      id: json['id'],
      consumerName: json['consumerName'],
      items: (json['items'] as List)
          .map((i) => DebtItem.fromJson(i))
          .toList(),
      totalAmount: (json['totalAmount'] as num).toDouble(),
      status: json['status'],
      createdAt: DateTime.parse(json['createdAt']),
      branchId: json['branch'] != null ? json['branch']['id'] : null,
      businessId: json['business'] != null ? json['business']['id'] : null,
    );
  }
}

class DebtItem {
  final String id;
  final String name;
  final String productCode;
  final int quantity;
  final double price;

  DebtItem({
    required this.id,
    required this.name,
    required this.productCode,
    required this.quantity,
    required this.price,
  });

  factory DebtItem.fromJson(Map<String, dynamic> json) {
    return DebtItem(
      id: json['id'],
      name: json['name'],
      productCode: json['productCode'],
      quantity: json['quantity'] as int,
      price: (json['price'] as num).toDouble(),
    );
  }
}

File: src/main/java/com/campbell/Flip/entities/Debt.java
package com.campbell.Flip.entities;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "debts")
public class Debt {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String consumerName;

    @OneToMany(mappedBy = "debt", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DebtItem> items;

    private double totalAmount;

    @Enumerated(EnumType.STRING)
    private DebtStatus status;

    private LocalDateTime createdAt;
    
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "branch_id")
    private Branch branch;

    @ManyToOne
    @JoinColumn(name = "business_id")
    private Business business;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (status == null) {
            status = DebtStatus.PENDING;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    // Getters and Setters
    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getConsumerName() {
        return consumerName;
    }

    public void setConsumerName(String consumerName) {
        this.consumerName = consumerName;
    }

    public List<DebtItem> getItems() {
        return items;
    }

    public void setItems(List<DebtItem> items) {
        this.items = items;
    }

    public double getTotalAmount() {
        return totalAmount;
    }

    public void setTotalAmount(double totalAmount) {
        this.totalAmount = totalAmount;
    }

    public DebtStatus getStatus() {
        return status;
    }

    public void setStatus(DebtStatus status) {
        this.status = status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    public Branch getBranch() {
        return branch;
    }

    public void setBranch(Branch branch) {
        this.branch = branch;
    }

    public Business getBusiness() {
        return business;
    }

    public void setBusiness(Business business) {
        this.business = business;
    }
}
